From cb123d671142cd78982ed3a43dd01bb3aea5c5d4 Mon Sep 17 00:00:00 2001
From: w30063796 <w30063796@notesmail.huawei.com/>
Date: Thu, 19 Sep 2024 17:14:31 +0800
Subject: [PATCH] TicketNo:DTS2024082129261 Description:optimized ipc
 useinnerAPI Team:OTHERS Feature or Bugfix:Bugfix Binary Source:No
 PrivateCode(Yes/No):No

Change-Id: I44287e2e519e6611bb3f40376487b85521b637d7
---
 pppd/chap.c          |  2 +-
 pppd/ipc_load_sa.cpp | 21 -------------
 pppd/ipc_load_sa.h   | 36 ----------------------
 pppd/ipc_request.cpp | 84 ++--------------------------------------------------
 4 files changed, 4 insertions(+), 139 deletions(-)
 delete mode 100644 pppd/ipc_load_sa.cpp
 delete mode 100644 pppd/ipc_load_sa.h

diff --git a/pppd/chap.c b/pppd/chap.c
index fa0574f..a7b23a3 100644
--- a/pppd/chap.c
+++ b/pppd/chap.c
@@ -573,7 +573,7 @@ chap_handle_status(struct chap_client_state *cs, int code, int id,
 	if (code == CHAP_SUCCESS) {
 		auth_withpeer_success(0, PPP_CHAP, cs->digest->code);
 #ifdef SUPPORT_SYSVPN
-		WriteStageAndStatus("pppdstart", 100);
+		WriteStageAndStatus("pppdstart", 0);
 #endif // SUPPORT_SYSVPN
 	} else {
 		cs->flags |= AUTH_FAILED;
diff --git a/pppd/ipc_load_sa.cpp b/pppd/ipc_load_sa.cpp
deleted file mode 100644
index 4bddc29..0000000
--- a/pppd/ipc_load_sa.cpp
+++ /dev/null
@@ -1,21 +0,0 @@
-/*
- * Copyright (c) 2024 Huawei Device Co., Ltd.
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-#include "ipc_load_sa.h"
-
-using namespace std;
-
-OnDemandLoadCertManagerCallback::OnDemandLoadCertManagerCallback(string servers) : servers(servers)
-{
-}
diff --git a/pppd/ipc_load_sa.h b/pppd/ipc_load_sa.h
deleted file mode 100644
index 4590ea8..0000000
--- a/pppd/ipc_load_sa.h
+++ /dev/null
@@ -1,36 +0,0 @@
-/*
- * Copyright (c) 2024 Huawei Device Co., Ltd.
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-#ifndef IPC_LOAD_SA_H
-#define IPC_LOAD_SA_H
-
-#include <future>
-#include <string>
-#include "iremote_object.h"
-#include "system_ability_load_callback_stub.h"
-
-class OnDemandLoadCertManagerCallback : public OHOS::SystemAbilityLoadCallbackStub {
-public:
-    OnDemandLoadCertManagerCallback(std::string servers);
-    void OnLoadSystemAbilitySuccess(int32_t systemAbilityId,
-        const OHOS::sptr<IRemoteObject>& remoteObject) override;
-    void OnLoadSystemAbilityFail(int32_t systemAbilityId) override;
-    OHOS::sptr<IRemoteObject> Promise(void);
-private:
-    std::string servers;
-    std::promise<OHOS::sptr<IRemoteObject>> promise_;
-};
-
-#endif /* IPC_LOAD_SA_H */
\ No newline at end of file
diff --git a/pppd/ipc_request.cpp b/pppd/ipc_request.cpp
index 4c97b57..23c2c09 100644
--- a/pppd/ipc_request.cpp
+++ b/pppd/ipc_request.cpp
@@ -14,90 +14,12 @@
  */
 
 #include "ipc_request.h"
-
-#include <chrono>
-#include <string>
-#include <thread>
-
-#include "securec.h"
-#include "ipc_load_sa.h"
-#include "iservice_registry.h"
-#include "system_ability_definition.h"
+#include "networkvpn_client.h"
 
 using namespace std;
-using namespace OHOS;
-
-namespace {
-    constexpr int SA_ID_VPN_SERVICE = COMM_VPN_MANAGER_SYS_ABILITY_ID;
-    constexpr int CMD_NOTIFY_CONNECT_STAGE = 14;
-    constexpr uint32_t MAX_SA_BOOT_DELAY_TIME = 30;
-    const std::u16string SA_VPN_SERVICE_DESCRIPTOR = u"OHOS.NetManagerStandard.INetworkVpnService";
-}
-
-static sptr<IRemoteObject> ProcessLoadSystemAbility(void)
-{
-    auto saManager = SystemAbilityManagerClient::GetInstance().GetSystemAbilityManager();
-    if (saManager == nullptr) {
-        return {};
-    }
-
-    auto object = saManager->CheckSystemAbility(SA_ID_VPN_SERVICE);
-    if (object != nullptr) {
-        return object;
-    }
-
-    string servers = "NetManager";
-    sptr<OnDemandLoadCertManagerCallback> loadCallBack = new (std::nothrow)OnDemandLoadCertManagerCallback(servers);
-    if (loadCallBack == nullptr) {
-        return {};
-    }
-
-    int32_t ret = saManager->LoadSystemAbility(SA_ID_VPN_SERVICE, loadCallBack);
-    if (ret != ERR_OK) {
-        return {};
-    }
-
-    return loadCallBack->Promise();
-}
+using namespace OHOS::NetManagerStandard;
 
 int32_t SendRequest(char *stage, int32_t state)
 {
-    uint32_t i = 0;
-    sptr<IRemoteObject> targetProxy = ProcessLoadSystemAbility();
-    while ((targetProxy == nullptr) && i < MAX_SA_BOOT_DELAY_TIME) {
-        std::this_thread::sleep_for(std::chrono::milliseconds(100)); /* 100 is time */
-        i++;
-    }
-
-    targetProxy = ProcessLoadSystemAbility();
-    if (targetProxy == nullptr) {
-        return -1;
-    }
-
-    MessageParcel data;
-    MessageParcel reply;
-    MessageOption option = MessageOption::TF_SYNC;
-
-    data.WriteInterfaceToken(SA_VPN_SERVICE_DESCRIPTOR);
-    data.WriteString(stage);
-    data.WriteInt32(state);
-
-    int error = targetProxy->SendRequest(static_cast<uint32_t>(CMD_NOTIFY_CONNECT_STAGE), data, reply, option);
-    return error;
-}
-
-void OnDemandLoadCertManagerCallback::OnLoadSystemAbilitySuccess(int32_t systemAbilityId,
-    const sptr<IRemoteObject> &remoteObject)
-{
-    promise_.set_value(remoteObject);
-}
-
-void OnDemandLoadCertManagerCallback::OnLoadSystemAbilityFail(int32_t systemAbilityId)
-{
-    promise_.set_value(nullptr);
-}
-
-sptr<IRemoteObject> OnDemandLoadCertManagerCallback::Promise(void)
-{
-    return promise_.get_future().get();
+    return NetworkVpnClient::GetInstance().NotifyConnectStage(std::string(stage), state);
 }
-- 
2.7.4

